src/Simplex_tree/concept/FiltrationValue.h

```diff
@@ -5,19 +5,102 @@
  *    Copyright (C) 2014 Inria
  *
  *    Modification(s):
+ *      - 2024/08 Hannah Schreiber: Update of the concept after several additions to the Simplex tree.
  *      - YYYY/MM Author: Description of the modification
  */
 
 /** \brief Value type for a filtration function on a cell complex.
-  *
-  * A <EM>filtration</EM> of a cell complex (see FilteredComplex) is
-  * a function \f$f:\mathbf{K} \rightarrow \mathbb{R}\f$ satisfying \f$f(\tau)\leq 
-  * f(\sigma)\f$ whenever \f$\tau \subseteq \sigma\f$. Ordering the simplices 
-  * by increasing filtration values (breaking ties so as a simplex appears after 
-  * its subsimplices of same filtration value) provides an indexing scheme 
-  * (see IndexingTag).
-  */
-  struct FiltrationValue {
-    /** \brief Operator < is a StrictWeakOrdering. */
-    bool operator<(FiltrationValue f1, FiltrationValue f2);
-  };
+ *
+ * Needs to implement `std::numeric_limits<FiltrationValue>::has_infinity`,
+ * `std::numeric_limits<FiltrationValue>::infinity()` and `std::numeric_limits<FiltrationValue>::max()`.
+ * But when `std::numeric_limits<FiltrationValue>::has_infinity` returns `true`,
+ * `std::numeric_limits<FiltrationValue>::max()` can simply throw when called, as well as,
+ * `std::numeric_limits<FiltrationValue>::infinity()` if `std::numeric_limits<FiltrationValue>::has_infinity`
+ * returns `false`.
+ *
+ * A <EM>filtration</EM> of a cell complex (see FilteredComplex) is
+ * a function \f$f:\mathbf{K} \rightarrow \mathbb{R}\f$ satisfying \f$f(\tau)\leq
+ * f(\sigma)\f$ whenever \f$\tau \subseteq \sigma\f$. Ordering the simplices
+ * by increasing filtration values (breaking ties so as a simplex appears after
+ * its subsimplices of same filtration value) provides an indexing scheme
+ * (see IndexingTag).
+ */
+struct FiltrationValue {
+  /**
+   * @brief Has to construct the default value of FiltrationValue.
+   */
+  FiltrationValue();
+  /**
+   * @brief Has to be able to take `0` as input at construction and in this case, construct an empty object.
+   * E.g., 0 for a numerical value or {} for a vector.
+   */
+  FiltrationValue(Any_arithmetic_type v);
```

Comment:
I already said that elsewhere, but I am not fond of this. The notion that FiltrationValue() will be 0 for 1-parameter and -inf for multiparameter while FiltrationValue(0) will be 0 for 1-parameter and *empty* for multiparameter is confusing.
Would `FiltrationValue(empty_value_t);` (loosely inspired by `std::nullopt_t`) look ok to you? `empty_value_t` could have a template conversion operator so by default it converts to T(0), but at least the uses would look more natural I think.

Reply:
I understand that it is a bit confusing, but I am not sure how we can be more consistent with the default values?
But even without multi-persistence, right now, `rec_deserialize` will fail in debug mode, if we do not explicitly put the value to 0. So we need either to document its behavior, or we should use an additional `if` to avoid constructing a Node with a filtration value when `store_filtration` is false. Which is probably the better solution here?
I am by the way a bit surprise that the problem does not arise at other places too. I guess, it's the precision randomness of `float`s...

Reply:
IIRC, what I had in mind was closer to
```c++
struct empty_filtration_value_t {
  template <class T> operator T() const { return T(0); }
} empty_filtration_value;
```
And users could add a constructor `MyType(empty_filtration_value_t)` to the definition of their custom filtration value type.
But 1. I may misremember and 2. I don't know if it is the best idea.

I promise this PR is still very high on my TODO-list, just below teaching...

Reply:
Could this method bring some ambiguity problems if the custom filtration value has several constructors which are not explicit? Or will the constructor always privilege `T = Filtration_value` if we do `Filtration_value f = empty_filtration_value;`?